ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PeerTransactionTracker.java

```diff
@@ -122,13 +123,14 @@ boolean hasPeerSeenTransaction(final EthPeer peer, final Hash txHash) {
   }
 
   private <T> Set<T> createTransactionsSet() {
-    return Collections.newSetFromMap(
-        new LinkedHashMap<>(1 << 4, 0.75f, true) {
-          @Override
-          protected boolean removeEldestEntry(final Map.Entry<T, Boolean> eldest) {
-            return size() > MAX_TRACKED_SEEN_TRANSACTIONS;
-          }
-        });
+    return Collections.synchronizedSet(
+        Collections.newSetFromMap(
+            new LinkedHashMap<>(1 << 4, 0.75f, true) {
```

Comment:
Honestly I haven't looked at it, but your point make sense, not sure about which could be a good initial size value, since this heavily depends on the network, for example I am pretty sure on mainnet we are reaching the max size soon or later, but on private networks with low traffic the size could grow very slowly. What do you suggest?

Reply:
I think the safest answer is to continue to set it to 16 (as opposed to 1 << 4). Any other number would ultimately be a guess.